src/SelfGravity/Poi_UserWorkBeforePoisson_GREP.cpp

```diff
@@ -244,68 +296,70 @@ void Combine_GREP_Profile( Profile_t *Prof[][2], const int lv, const int Sg, con
 {
 
    Profile_t *Prof_NonLeaf = Prof[NLEVEL][Sg];
+   Profile_t *Prof_Leaf;
 
 
-// combine the contributions from leaf patches on level <= lv and non-leaf patches on level = lv,
-// and store the sum in 'Prof_NonLeaf'
-   if ( Do_TEMPINT_in_ComputeProfile )
+// multiply the stored data by weight to reduce round-off errors
+   for (int b=0; b<Prof_NonLeaf->NBin; b++)
    {
-//    temporal interpolation is done in Aux_ComputeProfile()
-      Profile_t *Prof_Leaf = Prof[lv][Sg];
+      if ( Prof_NonLeaf->NCell[b] != 0L )   Prof_NonLeaf->Data[b] *= Prof_NonLeaf->Weight[b];
+   }
 
-      for (int b=0; b<Prof_Leaf->NBin; b++)
-      {
-         if ( Prof_Leaf->NCell[b] == 0L )  continue;
 
-         Prof_NonLeaf->Data  [b]  = Prof_NonLeaf->Weight[b] * Prof_NonLeaf->Data[b]
-                                  + Prof_Leaf   ->Weight[b] * Prof_Leaf   ->Data[b];
-         Prof_NonLeaf->Weight[b] += Prof_Leaf   ->Weight[b];
-         Prof_NonLeaf->NCell [b] += Prof_Leaf   ->NCell [b];
+// combine the contributions from the leaf and non-leaf patches on level = lv
+   Prof_Leaf = Prof[lv][Sg];
 
-         Prof_NonLeaf->Data  [b] /= Prof_NonLeaf->Weight[b];
-      }
+   for (int b=0; b<Prof_Leaf->NBin; b++)
+   {
+      if ( Prof_Leaf->NCell[b] == 0L )  continue;
+
+      Prof_NonLeaf->Data  [b] += Prof_Leaf->Data  [b] * Prof_Leaf->Weight[b];
+      Prof_NonLeaf->Weight[b] += Prof_Leaf->Weight[b];
+      Prof_NonLeaf->NCell [b] += Prof_Leaf->NCell [b];
    }
 
-   else
+
+// combine the contributions from the leaf patches on level < lv with temporal interpolation
+   for (int level=0; level<lv; level++)
    {
-//    apply temporal interpolation to leaf patches on level <= lv
-      for (int level=0; level<=lv; level++)
+      bool FluIntTime;
+      int  FluSg, FluSg_IntT;
+      int  Sg_Lv = GREPSg[level];
+      real FluWeighting, FluWeighting_IntT;
+
+      SetTempIntPara( level, Sg_Lv, PrepTime, GREPSgTime[level][Sg_Lv], GREPSgTime[level][1 - Sg_Lv],
+                      FluIntTime, FluSg, FluSg_IntT, FluWeighting, FluWeighting_IntT );
+
+                 Prof_Leaf      = Prof[level][FluSg];
+      Profile_t *Prof_Leaf_IntT = ( FluIntTime ) ? Prof[level][FluSg_IntT] : NULL;
+
+      for (int b=0; b<Prof_Leaf->NBin; b++)
       {
-//       temporal interpolation parameters
-         bool FluIntTime;
-         int  FluSg, FluSg_IntT;
-         real FluWeighting, FluWeighting_IntT;
+//       note that when GREP_OPT_FIXUP is disabled
+//       the NCell in Prof_Leaf and Prof_Leaf_IntT could be different due to patch allocation/deallocation
```

Comment:
This statement no longer holds since we always recompute profiles after grid refinement, right?

Reply:
Yes, Removed.